### PR TITLE
Added AVL tree implementation and updated corresponding readme.md

### PR DESCRIPTION
**Pull request by imsushant12**
Added the implementation of AVL tree and the related readme.md file 

ASSIGNED ISSUE : https://github.com/Algo-Phantoms/Algo-Tree/issues/236

### CODE REVIEW

File: Code/C++/avl_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+            }
+
+        case 5:
+            {
+                exit(0);
+                break;
+            }
+
+        default:
+            cout<<"\nINVALID CHOICE\n";
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:352]: add spaces around "<<" everywhere in code.

File: Code/C++/avl_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+                    cout<<"\nTree is not AVL";
+                break;
+            }
+
+        case 5:
+            {
+                exit(0);
+                break;
+            }
+
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:350]: remove extra lines like this.

File: Code/C++/avl_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+    int left = checkAVL(root->left);
+    int right = checkAVL(root->right);
+
+    if(BalanceFactor(root) >= -1 && BalanceFactor(root) <= 1)
+        return 1;
+
+    return 0;
+}
+
+
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:282]: remove 2 lines

File: Code/C++/avl_tree.cpp [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+            return(root->right);
+
+        else                                                                     //both left and left is null
+            return NULL;
+    }
+
+    if(root == NULL)
+        return root;
+
+    root->height = 1+(max(height(root->left) , height(root->right)));
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:250]: add spaces around "+" operator

File: Code/C++/avl_tree.cpp [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+
+    if(root->right == NULL)
+        return root;
+
+    else
+        maximumnode(root->right);
+}
+
+//deletes an element from tree
+node* deleteelement(node *root , int data)
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:217]: it would be better to name it as delete_element

File: Code/C++/avl_tree.cpp [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+
+    if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==1)
+        return RLrotation(root);
+
+    return root;
+}
+
+//return the minimum node of the tree
+node* minimumnode(node *root)
+{
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:192]: maximum_node.

File: Code/C++/avl_tree.cpp [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+    if(BalanceFactor(root)==2 && BalanceFactor(root->left)==1)
+        return LLrotation(root);
+
+    if(BalanceFactor(root)==2 && BalanceFactor(root->left)==-1)
+        return LRrotation(root);
+
+    if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==-1)
+        return RRrotation(root);
+
+    if(BalanceFactor(root)==-2 && BalanceFactor(root->right)==1)
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:184]: space around ==

File: Code/C++/avl_tree.cpp [hunk 8]
```diff
@@ -0,0 +1,429 @@
+#include <bits/stdc++.h>
```
Comment (rish-singhal) [inline Code/C++/avl_tree.cpp:1]: add description of the algorithm at the top.

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Code/C++/avl_tree.cpp (hunk 1 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 2 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 3 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 4 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 5 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 6 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp:199]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 7 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp]: done sir, please review

File: Code/C++/avl_tree.cpp (hunk 8 above)
Comment (imsushant12) [reply Code/C++/avl_tree.cpp:8]: done sir, please review

**Comment by rish-singhal** [comment]
Ready to get merged just resolve conflicts.

**Pull request merged**